### DISCUSSION TITLE
GZIP support for JSONInput ?

### DISCUSSION THREAD

**Discussion by AlexDut** [bentoml/BentoML#1007] [General] [answered]
Hi,

I was wondering if you plan to support gzip decoding for JSON data that was gzipped and sent to the server with `Content-Encoding: gzip` header ?

For the moment this seems not to be supported:
```bash
$ echo '{"text": "blablabla"}]' | gzip | curl -X POST -H --data-binary @- -H "Content-Encoding: gzip" my-bentoml-server:5000/predict

>> {"message":"BentoService error handling API request: Request content-type must be 'application/json' for this BentoService API"}
```

I guess this could be useful to reduce the size of large payloads (such a text data) sent over the network and thus would reduce the bandwidth load.

Thank you for your answer !

**Comment by parano** [comment]
Hi @AlexDut! Great question and I don't think BentoML current support gzip encoding, and we should definitely add support for it.

@bojiang @akainth015 this looks like something we want to consider in the new adapter refactoring that you guys are working on?

**Comment by bojiang** [comment, marked as answer]
Sounds great! 
So I drafted an PR on the dev branch, as well as the unit test:
https://github.com/bentoml/BentoML/pull/1015/commits/e07cfcde18ef63e4434411656c07fdf994f95c18

@parano Maybe we need to make it a generic method in order to implement it in each adapter?

**Reply by parano** [reply]
@bojiang yes that sounds great!

**Comment by thoo** [comment]
For the future reference,#1015 added this feature.
